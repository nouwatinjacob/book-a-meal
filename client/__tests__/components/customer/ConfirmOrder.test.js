import React from 'react';
import {
  shallow
} from 'enzyme';
import {
  ConfirmOrder,
  mapStateToProps
} from '../../../src/components/customer/ConfirmOrder.jsx';
import {
  mealResponse
} from '../../__mockData__/mockMeal';

const props = {
  getAMealAction: jest.fn(),
  makeOrderAction: () => Promise.resolve(),
  mealState: {
    meal: mealResponse.meal
  },
  orderState: {
    order: {
      quantity: 3
    },
    error: {
      response: {
        data: {
          message: 'Error messages'
        }
      }
    }
  }
};

const ids = [1, 2, 3];

describe('ConfirmOrder component', () => {
  localStorage.setItem('token', customerToken);
  sessionStorage.setItem('ids', ids);
  const setup = () => shallow(<ConfirmOrder {...props} />);

  it('should render ConfirmOrder component correctly', (done) => {
    const wrapper = setup();
    expect(wrapper).toMatchSnapshot();

    done();
  });

  it(`should call 'handleQuantityChange'
  when quantity is changed`, (done) => {
    const wrapper = setup();

    const event = {
      currentTarget: {
        value: '5'
      },
      preventDefault: jest.fn()
    };

    const handleQuantityChangeSpy = jest.spyOn(
      wrapper.instance(),
      'handleQuantityChange'
    );
    wrapper.instance().handleQuantityChange(event);
    expect(handleQuantityChangeSpy).toHaveBeenCalled();

    done();
  });

  it(`should call 'handleMakeOrder'
  when confirm order button is clicked`, (done) => {
    const wrapper = setup();

    const event = {
      preventDefault: jest.fn()
    };

    const handleMakeOrderSpy = jest.spyOn(
      wrapper.instance(),
      'handleMakeOrder'
    );
    wrapper.instance().handleMakeOrder(event);
    expect(handleMakeOrderSpy).toHaveBeenCalled();

    done();
  });
});