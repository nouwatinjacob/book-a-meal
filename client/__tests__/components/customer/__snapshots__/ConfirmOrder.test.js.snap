// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ConfirmOrder component should render ConfirmOrder component correctly 1`] = `
<div
  className="container"
>
  <CatererHeader />
  <div
    className="wrapper"
  >
    <div
      className="reduced-container"
    >
      <div
        className="row"
      >
        <div
          className="c-medium-6 c-small-12 c-xsmall-12"
          id="pd-0"
        >
          <h3>
            CART
          </h3>
        </div>
        <div
          className="c-medium-6 c-small-12 c-xsmall-12 amount"
          id="pd-0"
        >
          <button
            className="button default"
          >
            <Link
              replace={false}
              to="/menus"
            >
              Back to Menu
            </Link>
          </button>
        </div>
      </div>
      <br />
      <table>
        <tbody>
          <tr>
            <th />
            <th>
              Meal
            </th>
            <th>
              Quantity
            </th>
            <th>
              Unit Price(₦)
            </th>
            <th>
              Total Price(₦)
            </th>
          </tr>
          <tr>
            <td>
              <img
                alt=""
                height="100"
                src="https://res.cloudinary.com/sansaristic/image/upload/v1536197477/BookMeal/1536197474332pexels-photo-247685.png.png"
                width="100"
              />
            </td>
            <td>
              Fried Rice and chicken Lap
            </td>
            <td>
              <input
                max="20"
                min="1"
                name="quantity"
                onChange={[Function]}
                style={
                  Object {
                    "width": "100px",
                  }
                }
                type="number"
                value={1}
              />
            </td>
            <td>
              3000
            </td>
            <td>
              <strong>
                3000
              </strong>
            </td>
          </tr>
        </tbody>
      </table>
      <br />
      <div
        className="row"
      >
        <div
          className="c-medium-6 c-small-12 c-xsmall-12"
          id="pd-0"
        />
        <div
          className="c-medium-6 c-small-12 c-xsmall-12 amount"
          id="pd-0"
        >
          <button
            className="button warning"
            onClick={[Function]}
          >
            Confirm Order
          </button>
        </div>
      </div>
    </div>
  </div>
  <ToastContainer
    autoClose={5000}
    bodyClassName={null}
    className={null}
    closeButton={
      <CloseButton
        ariaLabel="close"
      />
    }
    closeOnClick={true}
    draggable={true}
    draggablePercent={80}
    hideProgressBar={false}
    newestOnTop={false}
    pauseOnHover={true}
    pauseOnVisibilityChange={true}
    position="top-right"
    progressClassName={null}
    rtl={false}
    style={null}
    toastClassName={null}
    transition={[Function]}
  />
</div>
`;
